Simplifications from a read-through

carterkozak opened the pull request:
==COMMIT_MSG==
Possible simplifications from a read-through
==COMMIT_MSG==

carterkozak committed:
Simplifications from a read-through

jkozlowski commented on dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutor.java:
```diff
@@ -54,49 +52,44 @@
             Thread.currentThread().interrupt();
 
-            abortQueue();
+            queue.poison();
```
Why do you think queue aborts are not necessary? You reckon aborting the main future will propagate everything correctly?

jkozlowski replied:
I think that's right, I haven't thought about that.

carterkozak replied:
Calling cancel on the FutureTask instances is equivalent to ignoring the queue. We can get away with it here because we're on the only thread that can drain the queue. I'm planning to add comments here describing the constraints.

jkozlowski commented on dialogue-blocking-channels/src/main/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutor.java:
```diff
@@ -54,49 +52,44 @@
+                try {
+                    toRun.run();
+                } catch (Throwable t) {
+                    log.error("Failed to execute runnable {}", toRun, t);
```
This should basically not happen, so I think we can have this here.

carterkozak replied:
Exactly, we submit a task which uses a SettableFuture as output rather than throwing exceptions outside of the Runnable.

carterkozak committed:
Remove QueueTake test utility in favor of setting Thread.interrupt

jkozlowski commented on dialogue-blocking-channels/src/test/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutorTest.java:
```diff
@@ -78,111 +81,140 @@ public void testThrowsWhenPoisoned() {
-        assertThat(Futures.getUnchecked(queueExecuted)).isTrue();
+            assertThat(Futures.getUnchecked(queueExecuted)).isTrue();
+        } finally {
+            shutdown(queueExecutor);
```
I've never bothered about shutting these in tests, but let's do it.

jkozlowski commented on dialogue-blocking-channels/src/test/java/com/palantir/dialogue/blocking/DefaultCallingThreadExecutorTest.java:
```diff
@@ -78,111 +79,135 @@ public void testThrowsWhenPoisoned() {
+
+            ListenableFuture<Boolean> queueExecuted = queueExecutor.submit(() -> {
+                try {
+                    Thread.currentThread().interrupt();
```
Ah, I totally missed this simpler way of doing this!

carterkozak committed:
safearg

carterkozak committed:
docs

carterkozak committed:
rename submit to execute

Since a future is no longer returned, this method matches
`Executor.execute(Runnable)`. Note that I don't implement/extend
`Executor` because this interface is not meant to be general
purpous.
